lib/matplotlib/axes/_axes.py

```diff
@@ -8748,14 +8806,15 @@ def violin(self, vpstats, positions=None, vert=None,
 
         # Render violins
         bodies = []
-        for stats, pos, width in zip(vpstats, positions, widths):
+        bodies_zip = zip(vpstats, positions, widths, facecolor)
+        for stats, pos, width, facecol in bodies_zip:
```

Comment:
```suggestion
        for stats, pos, width, facecolor in bodies_zip:
```
Minor, but we don't have to save letters 😄